### PR TITLE
cp: preserve permission on -p --parents

### LINKED ISSUE #4822: `cp -p --parents` should preserve permissions
Example:

```
mkdir -p a/b/c a/b/d e
touch a/b/c/foo a/b/d/foo
cp -p --parents a/b/c/foo e
chmod g-rx e/a e/a/b
dir=a
stat --printf %A $dir|sed s/s/x/g
echo ""
stat --printf %A e/$dir|sed s/s/x/g
```

tested by GNU in tests/cp/parent-perm.sh 
`bash -v util/run-gnu-test.sh tests/cp/parent-perm.sh`

### PR DESCRIPTION
**Pull request by shinhs0506**
fixes #4822

### CODE REVIEW

**Comment by sylvestre** [comment]
Good start, don't forget to add a test :)

File: src/uu/cp/src/cp.rs [hunk 1]
```diff
@@ -1753,6 +1759,25 @@ fn copy_helper(
...
+
+#[cfg(windows)]
+fn should_preserve_attribute(options: &Options) -> bool {
+    matches!(options.attributes.mode, Preserve::Yes { .. })
+        || matches!(options.attributes.timestamps, Preserve::Yes { .. })
+        || matches!(options.attributes.links, Preserve::Yes { .. })
+        || matches!(options.attributes.context, Preserve::Yes { .. })
+        || matches!(options.attributes.xattr, Preserve::Yes { .. })
+}
+
```
Comment (sylvestre) [inline src/uu/cp/src/cp.rs:1780]: what about?
```suggestion
fn should_preserve_attribute(options: &Options) -> bool {
    let checks = [
        &options.attributes.mode,
        &options.attributes.timestamps,
        &options.attributes.links,
        &options.attributes.context,
        &options.attributes.xattr,
    ];

    #[cfg(unix)]
    let checks = [
        checks,
        &options.attributes.ownership,
    ].concat();

    checks.iter().any(|attr| matches!(attr, Preserve::Yes { .. }))
}

```

Comment (shinhs0506) [reply src/uu/cp/src/cp.rs]: perfect! learning something new

**Comment by sylvestre** [comment]
Terrific, thanks :)

**Pull request merged**